Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue2577.cs

```diff
@@ -49,5 +47,41 @@ public DetailPage()
 				Content = button;
 			}
 		}
+
+#if UITEST
+		[Test]
+		public void Issue2577Test()
+		{
+			RunningApp.WaitForElement("NavigateButton");
+			RunningApp.Tap("NavigateButton");
+
+			RunningApp.WaitForElement("ToggleBackButton");
+
+			RunningApp.Screenshot("Hamburger menu icon is visible");
+
+			AppResult[] items = RunningApp.Query("OK");
+			Assert.AreNotEqual(items.Length, 0);
+			RunningApp.Tap("OK");
+
+			RunningApp.Screenshot("Flyout menu is showing");
+
+			RunningApp.SwipeRightToLeft();
+
+			RunningApp.Tap("ToggleBackButton");
+
+			items = RunningApp.Query("OK");
+			Assert.AreEqual(items.Length, 0);
+
+			RunningApp.Screenshot("Back arrow is showing");
+
+			var backArrow = RunningApp.Query(e => e.Class("AppCompatImageButton")).First(e => e.Label == null);
```

Comment:
Not particularly happy with how I need to access the back arrow here, but it works. Open to suggestions

Reply:
```suggestion
			var backArrow = RunningApp.Query(e => e.Class("Toolbar").Descendant("AppCompatImageButton")).Last();
```